core_backend/app/user_tools/routers.py

```diff
@@ -145,6 +134,56 @@ async def is_register_required(
     return RequireRegisterResponse(require_register=require_register)
 
 
+@router.put("/reset-password", response_model=UserRetrieve)
+async def reset_password(
+    user: UserResetPassword,
+    admin_user_db: Annotated[UserDB, Depends(get_admin_user)],
+    asession: AsyncSession = Depends(get_async_session),
+) -> UserRetrieve:
+    """
+    Reset password endpoint. Takes a user object, generates a new password,
+    replaces the old one in the database, and returns the updated user object.
+    """
+    try:
+        user_to_update = await get_user_by_username(
+            username=user.username, asession=asession
+        )
+
+        is_recovery_code_correct = user.recovery_code in user_to_update.recovery_codes
+
+        if not is_recovery_code_correct:
```

Comment:
suggest changing this line to just `if not user.recovery_code in user_to_update.recovery_codes:` and removing line `152`.